core/main/src/state/cap/permitted_state.rs

```diff
@@ -121,11 +122,24 @@ fn get_permissions_path(saved_dir: String) -> String {
 pub struct PermissionHandler;
 
 impl PermissionHandler {
+    fn get_distributor_alias_for_app_id(ps: &PlatformState, app_id: &str) -> String {
+        let dist_app_aliases = ps
+            .get_device_manifest()
+            .applications
+            .distributor_app_aliases;
+        if let Some(app_id_alias) = dist_app_aliases.get(&app_id.to_string()) {
+            app_id_alias.to_string()
+        } else {
+            app_id.to_string()
+        }
+    }
+
     pub async fn fetch_and_store(state: &PlatformState, app_id: &str) -> RippleResponse {
+        let app_id_alias = Self::get_distributor_alias_for_app_id(state, app_id);
         if state
             .cap_state
             .permitted_state
-            .get_app_permissions(app_id)
+            .get_app_permissions(app_id_alias.as_str())
```

Comment:
cap_state should store the permission with original app_id, not with the distributor alias. So the key should be app_id here.

Reply:
Replaced as suggested